directory/directory_test.go

```diff
@@ -0,0 +1,71 @@
+package directory
+
+import (
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCanonicalDockerReference(t *testing.T) {
+	dest := NewDirImageDestination("/path/to/somewhere")
+	_, err := dest.CanonicalDockerReference()
+	assert.NotNil(t, err)
+}
+
+func TestGetPutManifest(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "put-manifest")
+	assert.Nil(t, err)
```

Comment:
`assert.NoError(t, err)` (throughout)

and in cases like this, when failing would break the test, use `require` (which aborts on failure) instead of `assert` (which only marks the test as failed).